meta-mender-commercial/recipes-mender/mender-gateway/mender-gateway_git.bb

```diff
@@ -0,0 +1,27 @@
+require mender-gateway.inc
+
+inherit mender-closed-source-tar-utils
+
+SRCREV = "${@mender_closed_source_srcrev_from_src_uri(d, '${SRC_URI}', 'mender-gateway')}"
+
+def mender_gateway_version_from_preferred_version(d, srcrev):
+    pref_version = d.getVar("PREFERRED_VERSION")
+    if pref_version is not None and pref_version.find("-build") >= 0:
+        # If "-build" is in the version, use the version as is. This means that
+        # we can build tags with "-build" in them from this recipe, but not
+        # final tags, which will need their own recipe.
+        return pref_version
+    else:
+        # Else return "master${SRCREV}".
+        return "master%s" % srcrev
+
+PV = "${@mender_gateway_version_from_preferred_version(d, '${SRCREV}')}"
+
+# Override S with a wildcard
+S = "${WORKDIR}/mender-gateway-*"
```

Comment:
Woa! This one feels a bit risky... Doesn't the wildcard match `${PV}`? That's usually how it's defined...

Reply:
@kacf This seems to be the problem in CI. It works on my machine...

`PV` is not really reliable to use here, at least not with the current design. `PV` can be `master-git+86e78e1` or `master`. And for both cases directory inside the tarball is `mender-gateway-86e78e1`.

Can we have a short f2f to discuss options?

Reply:
Resolved f2f, the `master` package is now modified when copying it and here `S` becomes deterministic.